feature: Call Title and Subtitle

dhiraj opened the pull request:
## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* This PR adds support for Contoso to set a title and subtitle for a group call
* Settings activity has been modified to have two new text boxes to setup a call title and subtitle

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->

- [x] Yes
- [ ] No

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

<!-- Please check that applies to this PR using "x". -->

- [x] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [x] UI Changes
  - [x] Screen captures included
  - [x] Tested for Light/Dark mode
  - [x] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test
<!-- Add steps to run the tests suite and/or manually test -->

* Open the app
* Open settings
* Set up a title and subtitle in the new fields


## What to Check
Verify that the following are valid
* When a title is set, the call setup screen shows it, instead of "Setup"
* When a subtitle is set, the call setup screen shows it as expected

## Other Information
* This is being posted a public API change mainly because `CallCompositeLocalOptions` was modified to include two new `String` fields for the call title and subtitle
* Both Kotlin and Java launchers now _always_ send Local Options

## Screenshots
![Screenshot_1659647191](https://user-images.githubusercontent.com/43901/182954346-ef35f54a-b2c6-4c0c-ac93-67cb88593391.png)

![Screenshot_1659647182](https://user-images.githubusercontent.com/43901/182954374-9dc2e498-a23c-4f38-93b5-8460b0ef4fbf.png)

![Screenshot_1659647113](https://user-images.githubusercontent.com/43901/182954414-911874ce-613d-4f0b-89d9-2c5ae5381ef8.png)

![Screenshot_1659647147](https://user-images.githubusercontent.com/43901/182954464-efaa0149-1aab-4824-a527-9b8238105869.png)

pavelprystinka commented on azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/models/CallCompositeLocalOptions.java:
```diff
@@ -33,14 +35,21 @@
      * @see CallCompositeParticipantViewData
      */
-    public CallCompositeLocalOptions(final CallCompositeParticipantViewData participantViewData) {
+    public CallCompositeLocalOptions(final CallCompositeParticipantViewData participantViewData,
```
How about we will add a default constructor that does not take any parameters and add settert to all fields?

dhiraj replied:
@pavelprystinka  Fixed in d1ea113

Mrayyan reviewed (changes requested):
Tested looks good. 👍 

Please review with Alex P on what to do with cases like empty/blank title and subtitle. Min or max length for title and subtitle as well as what happens when an emoji is passed in.

Also, please update the changelog.

@pavelprystinka Do we first get this signed off by ARB and then merge it or is ARB after merge?

dhiraj commented:
> Please review with Alex P on what to do with cases like empty/blank title and subtitle. Min or max length for title and subtitle as well as what happens when an emoji is passed in.

When title is not specified we revert to current behavior and show "Setup" as the title. If a subtitle is specified, _without_ a title it is shown, under the current default title, Setup. I'll check with Alex P on min/max lengths for both the new fields. Thanks @Mrayyan 

(Screenshot with empty title and subtitle looks the same):
![Screenshot_1659732185](https://user-images.githubusercontent.com/43901/183160030-8730a74c-d23a-47da-808c-dbce63dbb197.png)

dhiraj commented:
@Mrayyan I've updated the CHANGELOG with a new Features section, please re-review, thanks!

Mrayyan reviewed (approved):
Looks good 👍

Mrayyan commented on azure-communication-ui/azure-communication-ui-demo-app/src/calling/java/com/azure/android/communication/ui/callingcompositedemoapp/launcher/CallingCompositeKotlinLauncher.kt:
```diff
@@ -38,7 +40,8 @@ class CallingCompositeKotlinLauncher(private val tokenRefresher: Callable<String
     ) {
         initialize(callLauncherActivity.applicationContext)
-        val participantViewData = getParticipantViewData(callLauncherActivity.applicationContext)
+        val
```
nitpicking here, you can bring this back to a single line.

dhiraj replied:
You're right, fixed in 8853acd

dhiraj commented:
After syncing up with iOS and UI design, I've updated the color of the main titlebar from full black (#000000) to almost black (#212121) and added a new color for subtitle as well. Updated screenshots:
![Screenshot_1660597375](https://user-images.githubusercontent.com/43901/184720242-441ba7fc-d007-4457-99dc-7702b456f5cc.png)
![Screenshot_1660597388](https://user-images.githubusercontent.com/43901/184720245-ee29e1d0-fb29-4d73-beba-108f02a1323d.png)

dhiraj commented:
Customized textSize styling on actionbar to reduce title and subtitle to match Figma mockups.

Updated Screenshots (note how subtitle accommodates more words now):
![Screenshot_1660685206](https://user-images.githubusercontent.com/43901/184989492-a40a1ca1-194a-4675-a427-e2f0f7a51915.png)
![Screenshot_1660685212](https://user-images.githubusercontent.com/43901/184989497-7d0d2850-a1d5-4de0-b329-353d85c98ec8.png)

phileo99 reviewed (approved):
👍
